app/cdap/components/RecipeList/ViewAllRecipes.tsx

```diff
@@ -14,32 +14,68 @@
  * the License.
  */
 
-import React from 'react';
+import React, { useState } from 'react';
 import T from 'i18n-react';
 import RecipeList from 'components/RecipeList';
-
 import Box from '@material-ui/core/Box';
 import { SortBy, SortOrder } from './types';
 import Breadcrumbs from '@material-ui/core/Breadcrumbs';
 import { Link } from 'react-router-dom';
 import Typography from '@material-ui/core/Typography';
 import { getCurrentNamespace } from 'services/NamespaceStore';
+import RecipeDetails from 'components/RecipeManagement/RecipeDetails';
+import { IRecipe } from 'components/RecipeList/types';
+import DrawerWidget from 'components/common/DrawerWidget';
+import RecipeHeaderActionTemplate from 'components/RecipeManagement/RecipeDetails/HeaderActionTemplate';
 
 const PREFIX = 'features.WranglerNewUI.Recipe';
 
 const redirectToObj = `/ns/${getCurrentNamespace()}/wrangle`;
 
+const VIEW_RECIPE = 'viewRecipe';
+
 const ViewAllRecipies = () => {
+  const [actionType, setActionType] = useState('');
+  const [isOpen, setIsOpen] = useState(false);
+  const [recipe, setRecipe] = useState<IRecipe>();
   const viewRecipeHandler = (selectedObject: any) => {
-    // To do : Integrate view recipe details panel
+    toggleOpen();
+    setRecipe(selectedObject);
+    setActionType(VIEW_RECIPE);
   };
 
   const handleEditRecipe = (selectedObject: any) => {
     // To do : Integrate Edit recipe details panel
   };
 
+  const toggleOpen = () => setIsOpen(!isOpen);
+
+  const renderRecipeDetailComponent = () => <RecipeDetails selectedRecipe={recipe} />;
+  const renderRecipeDetailHeaderActionTemplate = () => <RecipeHeaderActionTemplate />;
+  const renderEditRecipeComponent = () => <></>;
+
   return (
     <>
+      <DrawerWidget
+        anchor="right"
+        closeClickHandler={toggleOpen}
+        headingText={
+          actionType === VIEW_RECIPE
+            ? T.translate(`${PREFIX}.recipeDetails`)
+            : T.translate(`${PREFIX}.editRecipe`)
+        }
+        showBackIcon={false}
+        showDivider={actionType === VIEW_RECIPE ? true : false}
+        open={isOpen}
+        headerActionTemplate={
+          actionType === VIEW_RECIPE && renderRecipeDetailHeaderActionTemplate()
```

Comment:
why do we need to call this function here ? Can't we pass the `<RecipeHeaderActionTemplate />` JSX element without wrapping it in a function call ?

Reply:
removed wrapper function  and passed <RecipeHeaderActionTemplate />  directly